crates/task-impls/src/da.rs

```diff
@@ -107,8 +107,7 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 // the `DaProposalRecv` event. Otherwise, the view number subtraction below will
                 // cause an overflow error.
                 // TODO ED Come back to this - we probably don't need this, but we should also never receive a DAC where this fails, investigate block ready so it doesn't make one for the genesis block
-
-                if self.cur_view != TYPES::Time::genesis() && view < self.cur_view - 1 {
+                if self.consensus.read().await.cur_view() > proposal.data.view_number() + 1 {
```

Comment:
This seems to be changing the semantics (`self.cur_view` vs `consensus` view). Are we sure this is correct?

Reply:
I'll leave this ass is.  I'm pretty sure it's correct but probably not worth the change right now.  I want to protect agains the DA task falling behind consensus because of VID calculations but I think the other new check covers that